contracts/BIFI/strategies/Curve/StrategyConvex.sol

```diff
@@ -0,0 +1,375 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin-4/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin-4/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../../interfaces/common/IUniswapRouterETH.sol";
+import "../../interfaces/common/IWrappedNative.sol";
+import "../../interfaces/curve/ICurveSwap.sol";
+import "../../interfaces/curve/IGaugeFactory.sol";
+import "../../interfaces/curve/IRewardsGauge.sol";
+import "../Common/StratFeeManagerInitializable.sol";
+
+contract StrategyConvex is StratFeeManagerInitializable {
+    using SafeERC20 for IERC20;
+
+    // Tokens used
+    address public want; // curve lpToken
+    address public crv;
+    address public native;
+    address public depositToken;
+
+    // Third party contracts
+    address public gaugeFactory;
+    address public rewardsGauge;
+    address public pool;
+    uint public poolSize;
+    uint public depositIndex;
+    bool public useUnderlying;
+    bool public useMetapool;
+
+    // Routes
+    address[] public crvToNativeRoute;
+    address[] public nativeToDepositRoute;
+
+    struct Reward {
+        address token;
+        address[] toNativeRoute;
+        uint minAmount; // minimum amount to be swapped to native
+    }
+
+    Reward[] public rewards;
+
+    // if no CRV rewards yet, can enable later with custom router
+    bool public crvEnabled;
+    address public crvRouter;
+
+    // if depositToken should be sent as unwrapped native
+    bool public depositNative;
+
+    bool public harvestOnDeposit;
+    uint256 public lastHarvest;
```

Comment:
Should we change all lastHarvest to uint32? uint32 max number is sometime in 2108.